src/classes/BDI_ManageAdvancedMappingCtrl.cls

```diff
@@ -0,0 +1,719 @@
+        return enqueueMetadataRecords(new List<sObject>{customMetadata} );
+    }
+
+    /****************************************************************************************************
+    * @description Utility to convert a name into a valid 'DeveloperName' quality value (or unique GUID)
+    * for the CMT record
+    *
+    * @param name String
+    * @param makeUnique Boolean
+    * @return DeveloperName format string
+    */
+    public static String generateRecordName(String label) {
+        String name = '';
+
+        if (String.isNotBlank(label)) {
+            name = label.replace(' ', '_');
+        }
+
+        if (name.length() > 30) {
+            String firstFifteen = name.substring(0, 15);
+            String lastFifteen = name.substring(name.length() - 15, name.length());
+            name = firstFifteen + lastFifteen;
+        }
+
+        // Create a GUID format unique string
+        Blob b = Crypto.GenerateAESKey(128);
+        String h = EncodingUtil.ConvertTohex(b);
+        // force everything to have a 9-char random string at the end
+        name = name.left(30).removeEnd('_') + '_' + h.left(9);
+
+        name = name.replaceAll('[^\\w]+', '_').replaceAll('_{2,}', '_');
+        return name;
+    }
+
+    /*******************************************************************************************************
+    * @description Data_Import_Field_Mapping__mdt wrapper class used in the Field Mapping UI and for
+    * building corresponding custom metadata type records.
+    */
+    public class DataImportFieldMappingWrapper {
+        @AuraEnabled public String DeveloperName;
+        @AuraEnabled public String MasterLabel;
+        @AuraEnabled public String Label;
+        @AuraEnabled public String Maps_To_Icon;
+        @AuraEnabled public String Source_Field_Label;
+        @AuraEnabled public String Source_Field_API_Name;
+        @AuraEnabled public String Source_Field_Data_Type;
+        @AuraEnabled public String Target_Field_Label;
+        @AuraEnabled public String Target_Field_API_Name;
+        @AuraEnabled public String Target_Field_Data_Type;
+        @AuraEnabled public String Data_Import_Field_Mapping_Set;
+        @AuraEnabled public String Target_Object_Mapping;
+        @AuraEnabled public String Required;
+        @AuraEnabled public Boolean Is_Deleted;
+        @AuraEnabled public String Source_Field_Display_Type_Label;
+        @AuraEnabled public String Target_Field_Display_Type_Label;
+
+        public DataImportFieldMappingWrapper(Data_Import_Field_Mapping__mdt fieldMapping) {
+            String dataImport = 'npsp__DataImport__c';
+            String objectAPIName = fieldMapping.Target_Object_Mapping__r.Object_API_Name__c;
+            String sourceFieldAPIName = fieldMapping.Source_Field_API_Name__c;
+            String targetFieldAPIName = fieldMapping.Target_Field_API_Name__c;
+
+            if (NAMESPACE != 'npsp') {
```

Comment:
Hardcoding 'npsp' makes it difficult to test namespace handling by uploading a throwaway package with a different namespace.